packages/grafana-data/src/field/templateProxies.ts

```diff
@@ -0,0 +1,41 @@
+import { DataFrame, Field } from '../types';
+import { getFieldDisplayName } from './fieldState';
+import { formatLabels } from '../utils/labels';
+
+/**
+ * This object is created often, and only used when tmplates exist.  Using a proxy lets us delay
+ * calculations of the more complex structures (label names) until they are actually used
+ */
+export function getTemplateProxyForField(field: Field, frame?: DataFrame, frames?: DataFrame[]): any {
+  return new Proxy(
+    {} as any, // This object shows up in test snapshots
+    {
+      get: (obj: Field, key: string, reciever: any) => {
```

Comment:
@ryantxu I'm not super familiar with Proxy objects, but shouldn't the target here be the `field` passed to the `getTemplateProxyForField` function call? 🤯 Then the handler would use its first argument(`obj`) to reference the proxied object.